### PR TITLE
Upgrade dependencies, use npm exec, up node requirement

### PR DESCRIPTION
**Pull request by matt-boris**

### CODE REVIEW

**Comment by dstaley** [comment]
Hi @matt-boris! Feel free to ping me if you need any help with this PR. Whatever changes we make here will also need to be made against https://github.com/kinto/kinto.js so that we keep our tooling roughly in sync.

**Comment by matt-boris** [comment]
@dstaley I see you committed over here too and are having a similar issue. https://github.com/Kinto/kinto-http.js/pull/1164

I'll be working on figuring that out. If you figure anything out before you see any more commits here, just leave another comment please!

**Review by grahamalama — COMMENTED** [review]
I'm noticing some warnings in the console output when we `npm run test:chrome`:
```
(!) Plugin typescript: @rollup/plugin-typescript: Rollup requires that TypeScript produces ES Modules. Unfortunately your configuration specifies a "module" other than "esnext". Unless you know what you're doing, please change "module" to "esnext" in the target tsconfig.json file or plugin options.
(!) Plugin typescript: @rollup/plugin-typescript TS5055: Cannot write file '/home/runner/work/kinto-http.js/kinto-http.js/blob.js' because it would overwrite input file.
(!) Plugin replace: @rollup/plugin-replace: 'preventAssignment' currently defaults to false. It is recommended to set this option to `true`, as the next major version will default this option to `true`.
(!) Circular dependencies
<...>
(!) Use of eval is strongly discouraged
<...>
```

Do we know how long these warnings have been here? If they were a result of this PR, can we look into them for a bit and try and fix them? If they've been here for a while though, we can probably merge this and add an issue to remind ourselves to look into them.


Also, something I noticed while reviewing this PR: I don't think we need `setup-python` steps in our Actions since the `ubuntu-20.04` runner ships with Python 3.8. Not a necessary change for this PR, but possibly something we can remove in the future.

File: tsconfig.json [hunk 1]
```diff
@@ -16,8 +15,7 @@
     "resolveJsonModule": true,
     "experimentalDecorators": true,
     "downlevelIteration": true,
-    "typeRoots": ["./node_modules/@types", "./@types"],
-    "declaration": true
```
[the remark this answers is not in the document]
Comment (matt-boris) [reply tsconfig.json:20]: Quick snippet on `declaration` [here](https://www.typescriptlang.org/tsconfig#declaration).

And I assumed this is still needed. I moved it to [here](https://github.com/Kinto/kinto-http.js/pull/1183/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R15) which was suggested by a user https://github.com/rollup/plugins/issues/287#issuecomment-611368317 who had a similar issue to us in upgrading the `@rollup/plugin-typescript` lib.

File: .github/workflows/browsers.yml [hunk 2]
```diff
@@ -14,12 +14,26 @@ jobs:
...
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "16.x"
 
       - uses: actions/setup-python@v2
         with:
           python-version: "3.x"
 
+      - name: Upgrade Google Chrome
```
[the remark this answers is not in the document]
Comment (matt-boris) [reply .github/workflows/browsers.yml:27]: Nice 👀 Graham! I've gone ahead and created https://github.com/Kinto/kinto-http.js/issues/1190.

**Comment by matt-boris** [comment]
@grahamalama 
> Do we know how long these warnings have been here?

The most recent merged PR had all but two of the warnings: https://github.com/Kinto/kinto-http.js/runs/4072389130?check_suite_focus=true. The additional two warnings I believe just came from upgrading the `@rollup/plugin-typescript` lib and it having some stricter checks/warnings. I can look into them some more.

**Comment by matt-boris** [comment]
> If they were a result of this PR, can we look into them for a bit and try and fix them?

@grahamalama they've all been fixed!

**Review by grahamalama — APPROVED** [review]
> > If they were a result of this PR, can we look into them for a bit and try and fix them?
> 
> @grahamalama they've all been fixed!

💯 :shipit:

**Pull request merged**